third-party/re2/README

```diff
@@ -2,7 +2,7 @@
 re2 README for Chapel
 =====================
 
-This copy of re2 2017-04-01 (2b16c27) is being released with Chapel
+This copy of re2 2017-07-01 (a810d71) is being released with Chapel
```

Comment:
I know you're just upgrading RE2 but I'm not sure I'd say anymore that it's "released with Chapel for convenience" because we patch it to add functionality. In other words, you wouldn't get it to work if you replaced third-party/re2/re2-src with a checkout of RE2.
I think it made sense to say this in the past because it was third-party boilerplate and because we thought te RE2 project would eventually accept our patches. But I no longer think they will.